packages/core-modules/contracts/modules/CoreElectionModule.sol

```diff
@@ -180,8 +194,85 @@ contract CoreElectionModule is IElectionModule, ElectionStorage, OwnableMixin {
         electionData.addressVoted[msg.sender] = true;
     }
 
-    function getNomineeVotes(address nominee) external view override returns (uint) {
-        return _electionStore().electionData.nomineeVotes[nominee];
+    function isElectionEvaluated() external view virtual override returns (bool) {
+        return _electionStore().electionData.isElectionEvaluated;
+    }
+
+    function evaluateElectionBatch() external virtual override {
+        if (!isEpochFinished()) {
+            revert EpochNotFinished();
+        }
+
+        if (_electionStore().electionData.isElectionEvaluated) {
+            revert ElectionAlreadyEvaluated();
+        }
+
+        if (_electionStore().maxProcessingBatchSize == 0) {
+            revert BatchSizeNotSet();
+        }
+
+        _evaluateElectionBatchBySimpleCounting();
+    }
+
+    function _evaluateElectionBatchBySimpleCounting() internal virtual {
```

Comment:
So, from what I can see, this computation is split in batches configured in `store.maxProcessingBatchSize` correct?

Then the idea would be to reduce this value if an out-of-gas revert during the computation is observed. My question is: can this be adjusted during a computation that's already started?

Reply:
It can since each batch is processed independently and we store the offset of the latest index processed.
Also, most likely (I'd say for sure, but maybe I'm missing some nit of gas calculation) this will revert on the first attempt to process a batch if the batch size is too large, so technically it won't be adjusted when computation is already started.

Reply:
Sweet 💯